analytics-api/src/analytics_api/models/user_response_detail.py

```diff
@@ -80,10 +78,20 @@ def get_response_count_by_created_week(
                         extract('week', func.timezone('America/Vancouver', UserResponseDetail.created_date))
                         ).label('showdataby'),
             func.count(UserResponseDetail.id).label('responses'))
-            .filter(*queries)
+            .filter(*filters)
             .order_by('orderby')
             .group_by('showdataby', 'orderby').all()
         )
         cols = ['showdataby', 'responses']
         result = [{col: getattr(d, col) for col in cols} for d in response_count_by_created_week]
         return jsonify(result)
+
+    @staticmethod
+    def _append_search_options_filters(filters, search_options):
+        if search_options.get('from_date'):
+            filters.append(cast(UserResponseDetail.created_date, Date) >= search_options.get('from_date'))
+
+        if search_options.get('to_date'):
+            filters.append(cast(UserResponseDetail.created_date, Date) <= search_options.get('to_date'))
+
```

Comment:
i think we dont need cast..
also check if  if 'from_date' in search_options:

Reply:
updated the if condition